wire-ios-sync-engine/Tests/Source/Use cases/ImportBackupUseCaseTests.swift

```diff
@@ -0,0 +1,187 @@
+    private var dispatchGroup: ZMSDispatchGroup!
+    private var sharedContainerURL: URL!
+    private var mockUserSession: MockUserSession!
+    private var sut: ImportBackupUseCase!
+
+    override func setUp() async throws {
+        let fileManager = FileManager()
+        let selfUserQualifiedID = QualifiedID.random()
+
+        coreDataStack = try await CoreDataStackHelper()
+            .createStack(inMemoryStore: true)
+
+        mockStreamDecryptor = .init()
+        mockStreamDecryptor.decryptInputOutputAccountIDPassword_MockMethod = { _, _, _, _ in }
+
+        mockFileArchiver = .init()
+        mockFileArchiver.unzipFileAtTo_MockMethod = { _, _ in }
+
+        mockEntityStorage = .init()
+        mockEntityStorage.importsDirectory = fileManager
+            .temporaryDirectory
+            .appending(path: UUID().uuidString)
+        mockEntityStorage.createContextProviderAccountApplicationContainerDispatchGroup_MockMethod = { _, _, _ in
+            let stack = try await CoreDataStackHelper()
+                .createStack(inMemoryStore: true)
+            try await stack.viewContext.perform {
+                let user = ZMUser.selfUser(in: stack.viewContext)
+                user.remoteIdentifier = selfUserQualifiedID.uuid
+                user.domain = selfUserQualifiedID.domain
+                try stack.viewContext.save()
+            }
+            return stack
+        }
+        mockEntityStorage
+            .replacePersistentStoreAccountIdentifierFromApplicationContainerDispatchGroup_MockMethod = { _, _, _, _ in
+                URL(filePath: "/accountDataFolder/")
+            }
+
+        mockAppStateUpdater = .init()
+        mockAppStateUpdater.reportImportProgressProgress_MockMethod = { _ in }
+        mockAppStateUpdater.reportMigrationNeeded_MockMethod = {
+            self.coreDataStack = nil
+        }
+        mockAppStateUpdater.selectAccountAndTriggerSlowSync_MockMethod = { _ in }
+
+        dispatchGroup = .init(label: UUID().uuidString)
+
+        sharedContainerURL = fileManager
+            .temporaryDirectory
+            .appending(path: UUID().uuidString)
+
+        let viewContext = coreDataStack.viewContext
+        let selfUser = await viewContext.perform {
+            let user = ZMUser.selfUser(in: viewContext)
+            user.remoteIdentifier = selfUserQualifiedID.uuid
+            user.domain = selfUserQualifiedID.domain
+            return user
+        }
+
+        mockUserSession = .init()
+        mockUserSession.contextProvider = coreDataStack
+        mockUserSession.selfUser = selfUser
+        try await viewContext.perform { [viewContext] in
+            let selfUserClient = UserClient.insertNewSelfClient(
+                in: viewContext,
+                selfUser: selfUser,
+                model: "",
+                label: ""
+            )
+            selfUserClient.markAsSelfClient()
+            self.mockUserSession.selfUserClient = selfUserClient
```

Comment:
Perhaps we should have a test that asserts the self user and self user client are preserved after a backup has been imported.

Reply:
Added an assertion: b0a53f035590d86f2ab15ffcd86cb3d2c4b70503